e2e/instances/list/index.spec.ts

```diff
@@ -0,0 +1,388 @@
+    `instance-row-revision-id-${instance.instance}`
+  );
+  await expect(
+    revisionId,
+    "revision id should appear in the row"
+  ).toContainText("latest");
+
+  const invoker = page.getByTestId(`instance-row-invoker-${instance.instance}`);
+  await expect(
+    invoker,
+    "invoker should appear in the row, to be api for this instance"
+  ).toContainText("api");
+
+  const state = page.getByTestId(`instance-row-state-${instance.instance}`);
+
+  await expect(
+    state,
+    "state should appear in the row, to be same status from the api response"
+  ).toContainText(instanceDetail?.status.toString() || "pending");
+
+  if (instanceDetail?.status === "failed") {
+    await state.hover();
+    const errorTooltip = page.getByTestId(
+      `instance-row-state-error-tooltip-${instance.instance}`
+    );
+    await expect(
+      errorTooltip,
+      "on hover the failed badge, error tooltip should appear"
+    ).toBeVisible();
+  }
+
+  const createdReltime = page.getByTestId(
+    `instance-row-relative-created-time-${instance.instance}`
+  );
+  await expect(createdReltime, "createAt should be visible").toBeVisible();
+
+  await page
+    .getByRole("heading", { name: "Recently executed instances" })
+    .click(); // click on table header to close all tooltips opened
+  await createdReltime.hover({ force: true });
+  const createdAtTooltip = page.getByTestId(
+    `instance-row-absolute-created-time-${instance.instance}`
+  );
+  await expect(
+    createdAtTooltip,
+    "on hover, the absolute time should appear"
+  ).toBeVisible();
+
+  const updatedReltime = page.getByTestId(
+    `instance-row-relative-updated-time-${instance.instance}`
+  );
+  await expect(updatedReltime, "updateAd should be visible").toBeVisible();
+
+  await page
+    .getByRole("heading", { name: "Recently executed instances" })
+    .click(); // click on table header to close all tooltips opened
+  await updatedReltime.hover({ force: true });
+  const updatedAtTooltip = page.getByTestId(
+    `instance-row-absolute-updated-time-${instance.instance}`
+  );
+  await expect(
+    updatedAtTooltip,
+    "on hover, the absolute time should appear"
+  ).toBeVisible();
+
+  const workflowLink = page.getByTestId(
+    `instance-row-workflow-${instance.instance}`
+  );
+  await workflowLink.click();
+  await expect(
+    page,
+    "on click workflow, page should navigate to the workflow page"
+  ).toHaveURL(
+    `/${namespace}/explorer/workflow/active/${instanceDetail?.as.split(":")[0]}`
+  );
+
+  await page.goBack();
+  await instanceItemRow.click();
+  await expect(
+    page,
+    "on click row, page should navigate to the instance detail page"
+  ).toHaveURL(`/${namespace}/instances/${instance.instance}`);
+
+  //no pagination to be visibe
+  await expect(
+    page.getByTestId("instance-list-pagination"),
+    "there is no pagination when there aren't many instances"
+  ).not.toBeVisible();
+});
+
+test("test the pagination", async ({ page }) => {
```

Comment:
About this test in general, can you go over it and check if you can introduce some more variables that make the numbers more understandable. You could for example give `Math.floor(Math.random() * 12)` a more meaningful name. Even the numbers, `12`, `170` and `15` could be variables to make it easier to understand the context.